src/lxc/network.c

```diff
@@ -2679,6 +2717,151 @@ bool lxc_delete_network_unpriv(struct lxc_handler *handler)
 	return true;
 }
 
+struct ip_proxy_args {
+	const char *ip;
+	const char *dev;
+};
+
+static int lxc_add_ip_proxy_exec_wrapper(void *data)
+{
+	struct ip_proxy_args *args = data;
+
+	execlp("ip", "ip", "neigh", "add", "proxy", args->ip, "dev", args->dev,
+	       (char *)NULL);
+	return -1;
+}
+
+static int lxc_del_ip_proxy_exec_wrapper(void *data)
+{
+	struct ip_proxy_args *args = data;
+
+	execlp("ip", "ip", "neigh", "flush", "proxy", args->ip, "dev", args->dev,
+	       (char *)NULL);
+	return -1;
+}
+
+static int lxc_add_ip_proxy(const char *ip, const char *dev)
+{
+	int ret;
+	char cmd_output[PATH_MAX];
+	struct ip_proxy_args args;
+	args.ip = ip;
+	args.dev = dev;
+
+	ret = run_command(cmd_output, sizeof(cmd_output),
+			  lxc_add_ip_proxy_exec_wrapper, (void *)&args);
```

Comment:
Why the explicit cast to void?

Reply:
@brauner this is how it is used elsewhere in the code base. I think its because the function signature for `run_command` requires that way https://github.com/lxc/lxc/blob/master/src/lxc/utils.c#L1597 so that the same function can be used with commands that take different sets of arguments.

Reply:
@brauner I've tried removing the cast and it compiles, so I do wonder why it was added in the past:

https://github.com/lxc/lxc/blob/ac7148050126cc3cd9872007c5ea69cad44baa91/src/lxc/lsm/apparmor.c#L811
https://github.com/lxc/lxc/blob/ea84ddf9e2e4933bf8110366be3c3f6dd3c4b6a6/src/lxc/network.c#L2127-L2128

There's quite a few like that.